cmd/start/cmd.go

```diff
@@ -55,6 +58,17 @@ func startParsing(ctx *parser.Context) error {
 	cfg := ctx.Config.Parser
 	logging.StartHeight.Add(float64(cfg.StartHeight))
 
+	// Start the prometheus monitoring
+	monitoringCfg := ctx.Config.Monitoring
+	if monitoringCfg.Enabled {
+		http.Handle("/metrics", promhttp.Handler())
+		server := &http.Server{
+			Addr:              fmt.Sprintf(":%d", monitoringCfg.Port),
+			ReadHeaderTimeout: 3 * time.Second,
+		}
+		go server.ListenAndServe()
+	}
```

Comment:
Can we create a new `type` for this inside another package? Maybe something like `prometheus` that has a `Server` type with a `Start` and `Stop` method.

We should also make sure to stop this inside the `trapSignal` method